src/common/function/FunctionManager.cpp

```diff
@@ -2766,6 +2771,34 @@ FunctionManager::FunctionManager() {
       }
     };
   }
+  {
+    auto &attr = functions_["json_extract"];
+    // note, we don't support second argument(path) like MySQL JSON_EXTRACT for now
+    attr.minArity_ = 1;
+    attr.maxArity_ = 1;
+    attr.isAlwaysPure_ = true;
+    attr.body_ = [](const auto &args) -> Value {
+      if (!args[0].get().isStr()) {
+        return Value::kNullBadType;
+      }
+      auto json = args[0].get().getStr();
+
+      // invalid string to json will be caught and returned as null
+      try {
+        auto obj = folly::parseJson(json);
+        if (!obj.isObject()) {
```

Comment:
Why check this?

Reply:
Or invalid json string will end up abnormal here, led the connection hanging.

Reply:
`"string"` is not object but also valid json.

Reply:
Oh, thanks @Shylock-Hg , now I know "foo" is indeed a valid JSON according to RFC 8259, I will revisit here and fix this case!

Reply:
@Shylock-Hg , in SQL DB, "foo" was parsed as NULL, thus I think checking isObject() is still reasonable here, what do you think?

<img width="958" alt="Screen Shot 2022-10-24 at 10 19 17" src="https://user-images.githubusercontent.com/1651790/197435959-d195c299-62a8-403f-86a5-61c7389e5b21.png">

Reply:
aligned with @Shylock-Hg to go with option-1, I have updated the highlight in this PR to ensure this behavior will be emphasized in docs.